app/models/conversion_host.rb

```diff
@@ -21,31 +26,97 @@ class ConversionHost < ApplicationRecord
 
   include_concern 'Configurations'
 
-  # To be eligible, a conversion host must have the following properties
-  #  - A transport mechanism is configured for source (set by 3rd party)
-  #  - Credentials are set on the resource and SSH connection works
-  #  - The number of concurrent tasks has not reached the limit
+  # Check using all associated authentications if any are directly associated
+  # with the conversion host. Otherwise, use the default check which uses the
+  # associated resource's authentications.
+  #
+  # Subclasses should pass provider-specific +options+, such as proxy information.
+  #
+  # This method is necessary to comply with AuthenticationMixin interface.
+  #--
+  # TODO: Use the verify_credentials_ssh method in host.rb? Move that to the
+  # AuthenticationMixin?
+  #
+  def verify_credentials(_auth_type = nil, options = {})
+    if authentications.empty?
+      check_ssh_connection
+    else
+      require 'net/ssh'
+      host = hostname || ipaddress
+
+      authentications.each do |auth|
+        user = auth.userid || ENV['USER'] || Etc.getlogin
+
+        ssh_options = { :timeout => 10, :logger => $log, :verbose => :error }
+
+        if auth.password
+          ssh_options[:password] = auth.password
+          ssh_options[:auth_methods] = %w[password]
+        end
+
+        if auth.auth_key
```

Comment:
So if you need more than userid+password you can use `auth = authentication_type(auth_type)` instead of what I said above.

Reply:
Since this is an internal check, I don't think `auth_type` would ever be set, and I had a hard time finding examples in other code where it wasn't nil.

In practice, I doubt there would ever be more than one associated authentication, but I figured if more than one ever was set, they should all be checked.

Reply:
> Since this is an internal check, I don't think auth_type would ever be set, and I had a hard time finding examples in other code where it wasn't nil.

I don't think I've ever seen it not be set...

```
=> [["ManageIQ::Providers::Vmware::InfraManager dev-vc65", "default"], ["ManageIQ::Providers::Vmware::InfraManager::HostEsx dell-r430-15.cloudforms.lab.eng.rdu2.redhat.com", "default"], ["ManageIQ::Providers::Vmware::InfraManager::HostEsx dell-r430-16.cloudforms.lab.eng.rdu2.redhat.com", "default"], ["ManageIQ::Providers::Vmware::InfraManager::HostEsx dell-r430-17.cloudforms.lab.eng.rdu2.redhat.com", "default"]]
```

> In practice, I doubt there would ever be more than one associated authentication, but I figured if more than one ever was set, they should all be checked.

This is probably true so the authtype should be `"default"`

Reply:
Where/how does this get selected? Is this an external option coming from the UI? I guess I'm not sure how it fits here, since we're relying on the properties of the associated authentications to pass the proper options. Allowing an auth_type wouldn't be sensical to me. Perhaps I'm missing some context?

Reply:
Updated to use auth_type first, which will return a result if not nil.